UploanPH/unleash
# Uploan unleash implementation

## Overview
This pull request solves the following issue(s):
- [Secure Unleash API Client](https://uploan.atlassian.net/browse/ISVC-65)
- [Secure Unleash Admin Client](https://uploan.atlassian.net/browse/ISVC-66)

Create Uploan authentication hooks for Unleash's admin and client API endpoints

I have:

- [x] Tested this locally and reviewed my own code.
- [ ] Provided appropriate unit tests for my change.
- [x] Added a description of my changes.
- [x] Provided test instructions for the reviewer.

## Analysis
Unleash implementation for Uploan must be secured. This includes adding oauth implementation to Unleash endpoints. As stated on the [securing Unleash section](https://unleash.github.io/docs/securing_unleash), the way to add custom security to Unleash is to create authentication hooks.

## Acceptance Criteria
API endpoints of Unleash should be secured by using Uploan OAuth implementation.

## Test Instructions
After fetching and checking out this branch on your local environment:
1. use a client credentials to generate token bearer.
2. use the token bearer to request to Unleash endpoints
3. if the token bearer does not exist, you should received an unauthorized response.
4. If the token bearer exist and valid, you must successfully access the Unleash endpoints.

List of unleash endpoints:
See the attached Postman collections.
[Feature Flag Service.postman_collection.json.zip](https://github.com/UploanPH/unleash/files/5013706/Feature.Flag.Service.postman_collection.json.zip)




## Guides
For the development standards, coding best practices, and code review guidelines. Kindly refer to this document(s): 
- [Uploan Engineering Handbooks](https://uploan.atlassian.net/wiki/spaces/0T/pages/662340021/Handbooks)
